packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -184,73 +207,64 @@ const DialpadContainer = (props: {
   onSendDtmfTone?: (dtmfTone: DtmfTone) => Promise<void>;
   /**  Callback for dialpad button behavior */
   onClickDialpadButton?: (buttonValue: string, buttonIndex: number) => void;
-  /**  customize dialpad input formatting */
-  onDisplayDialpadInput?: (input: string) => string;
+  /** Pass in custom content to dialpad textfield */
+  textFieldValue?: string;
   /**  on change function for text field */
   onChange?: (input: string) => void;
   /**  boolean input to determine when to show/hide delete button, default true */
   showDeleteButton?: boolean;
   styles?: DialpadStyles;
 }): JSX.Element => {
   const theme = useTheme();
-  const [textValue, setTextValue] = useState('');
 
-  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput, onChange, showDeleteButton = true } = props;
+  const { onSendDtmfTone, onClickDialpadButton, textFieldValue, onChange, showDeleteButton = true } = props;
 
-  const sanitizeInput = (input: string): string => {
-    // remove non-valid characters from input: letters,special characters excluding +, *,#
-    return input.replace(/[^\d*#+]/g, '');
-  };
+  const [plainTextValue, setPlainTextValue] = useState(textFieldValue ?? '');
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(plainTextValue);
+    }
+  }, [plainTextValue, onChange]);
+
+  useEffect(() => {
+    setText(textFieldValue ?? '');
+  }, [textFieldValue]);
 
   const onClickDialpad = (input: string, index: number): void => {
-    // remove non-valid characters from input: letters,special characters excluding +, *,#
-    const value = sanitizeInput(textValue + input);
-    setTextValue(value);
+    setText(plainTextValue + input);
     if (onSendDtmfTone) {
       onSendDtmfTone(DtmfTones[index]);
     }
     if (onClickDialpadButton) {
       onClickDialpadButton(input, index);
     }
-    if (onChange) {
-      onChange(onDisplayDialpadInput ? onDisplayDialpadInput(value) : formatPhoneNumber(value));
-    }
   };
 
   const onLongPressDialpad = (input: string, index: number): void => {
-    let value;
     if (input === '0' && index === 10) {
-      // remove non-valid characters from input: letters,special characters excluding +, *,#
-      value = sanitizeInput(textValue + '+');
-      setTextValue(value);
+      setText(plainTextValue + '+');
```

Comment:
I might be forgetting something from this morning but is this supposed to be: 
`'+' + plainTextValue`? 

I feel like this would result in a phone number like `14443332222+`

Reply:
This is code for the longpress functionality, long pressing 0 will append + instead of 0